src/pal/src/sharedmemory/sharedmemory.cpp

```diff
@@ -920,14 +945,15 @@ void SharedMemoryProcessDataHeader::Close()
     }
 
     // Delete the shared memory file, and the session directory if it's not empty
-    char path[SHARED_MEMORY_MAX_FILE_PATH_CHAR_COUNT + 1];
-    SIZE_T sessionDirectoryPathCharCount = SharedMemoryHelpers::CopyString(path, 0, SHARED_MEMORY_SHARED_MEMORY_DIRECTORY_PATH);
-    path[sessionDirectoryPathCharCount++] = '/';
-    sessionDirectoryPathCharCount = m_id.AppendSessionDirectoryName(path, sessionDirectoryPathCharCount);
-    path[sessionDirectoryPathCharCount++] = '/';
-    SharedMemoryHelpers::CopyString(path, sessionDirectoryPathCharCount, m_id.GetName(), m_id.GetNameCharCount());
+    PathCharString path;
+    VerifyStringOperation(SharedMemoryManager::CopySharedMemoryBasePath(path)
```

Comment:
A nit - it seems that using VerifyStringOperation for each of the appends separately would be more consistent with the other places above. And it would be easier to debug too.

Reply:
And the same at few other places below.